aws-quickstart/taskcat
# exit when there are errors deleting objects from S3

## Overview

boto3's `s3.delete_objects` call does not raise an exception when there are errors deleting files, instead it returns an `Errors` key containing details for each file that failed.

This pr checks for that key, prints error messages and then raises a `TaskCatException`, which triggers `sys.exit(1)`.

Deleting a file that does not exist does not raise an exception, which is fine imo, as the desired state is is consistent.

## Testing/Steps taken to ensure quality

ran taskcat against bucket where policy denied delete. prints error and exits. also ran `delete_objects` against non existent files (does not error).

## Testing Instructions

1. run taskcat with `-u` to upload some files.
2. set a bucket policy denying `DeleteObject`
3. delete/rename a file locally
4. run taskcat again

It should exit with the following:

```
[INFO   ] :Staging Bucket => mybucket
[S3: DELETE ] s3://mybucket/prefix/test.txt
[ERROR  ] :S3 delete error: {'Key': 'prefix/test.txt', 'Code': 'AccessDenied', 'Message': 'Access Denied'}
[ERROR  ] :Failed to delete one or more files from S3
```
